frontend/src/app/favorite-topics/favorite-topics.component.ts

```diff
@@ -0,0 +1,88 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Category} from '../_models/category';
+import {AgeRange, Covid19Categories, Language} from '../_models/hit_type';
+import {FavoriteTopicsDialogComponent} from '../favorite-topics-dialog/favorite-topics-dialog.component';
+import {MatDialog} from '@angular/material/dialog';
+import {User} from '../_models/user';
+import {UserFavorite} from '../_models/user_favorite';
+import {ApiService} from '../_services/api/api.service';
+
+@Component({
+  selector: 'app-favorite-topics',
+  templateUrl: './favorite-topics.component.html',
+  styleUrls: ['./favorite-topics.component.scss']
+})
+export class FavoriteTopicsComponent implements OnInit {
+  @Input() currentUser: User;
+  favoriteTopics: Category[] = [];
+  favoriteAges: string[] = [];
+  favoriteLanguages: string[] = [];
+  favoriteCovid19Topics: string[] = [];
+
+  ageLabels = AgeRange.labels;
+  languageLabels = Language.labels;
+  covid19Labels = Covid19Categories.labels;
+
+  constructor(
+    private api: ApiService,
+    public dialog: MatDialog
+  ) { }
+
+  ngOnInit() {
+    this.loadFavorites();
+  }
+
+  loadFavorites() {
+    this.api.getFavoritesByUserAndType(this.currentUser, 'category').subscribe(favorites => {
```

Comment:
I think we could DRY this up, but it works for now -- making a note to circle back when time allows